core/src/main/java/io/fabric8/maven/core/config/MappingConfig.java

```diff
@@ -0,0 +1,24 @@
+package io.fabric8.maven.core.config;
+
+import org.apache.maven.plugins.annotations.Parameter;
+
+public class MappingConfig {
+
+    @Parameter(required = true)
+    private String kind;
```

Comment:
Just as a side note: For deeper objects in a plugin configuration, `@Parameter` is not evaluated but just reflection is used. `@Parameter` works only on top-level properties in a Mojo, so they are not evaluated here and should probably be removed.

Reply:
But I have seen that in all other examples they are doing in this way, I am not saying that you are not right, I'll check, and keep you updated.

Reply:
It seems you are right (https://maven.apache.org/plugin-tools/maven-plugin-annotations/apidocs/org/apache/maven/plugins/annotations/Parameter.html) then let me write an it test and check if it works or not.

Reply:
Yeah, I used that `@Parameter` also blindly before I found out that it doesn't change anything. And then forgot to clean up (as it doesn't hurt either)